### PR TITLE
Document pip support

### PR DESCRIPTION
**Pull request by chmeliik**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [n/a] Code coverage from testing does not decrease and new code is covered
- [n/a] New code has type annotations
- [x] Docs updated (if applicable)
- [x] Docs links in the code are still valid (if docs were updated)

### CODE REVIEW

**Comment by chmeliik** [comment]
Still missing an update to the Usage documentation, but I don't think I can fit that into this PR

Note that I left out some things that we do support, most notably the [#cachito_hash](https://github.com/containerbuildsystem/cachi2/blob/474c9993d77b7a66d71dc97bcf2d6ec585967c6d/cachi2/core/package_managers/pip.py#L1294) nonsense (we never should have supported it, we should have support [PEP 440 direct references](https://peps.python.org/pep-0440/#direct-references) style hashes)

**Comment by chmeliik** [comment]
Oh, and as always, you're probably better off reviewing the rendered docs rather than the markdown source https://github.com/chmeliik/cachi2/blob/document-pip/docs/pip.md

**Comment by brunoapimentel** [comment]
I just finished my first read, and amazing content as usual, Adam! :rocket: 
I will do a second read with `nitpicking: on`. 

For now, I'm wondering if we should have a full usage example based on pip (similar to what we have with [Go](https://github.com/containerbuildsystem/cachi2/blob/main/docs/usage.md)). Definitely not something to cover in this PR, though.

**Comment by lkolacek** [comment]
I finished my first reading and it looks great! Thank you :+1:

**Comment by chmeliik** [comment]
> For now, I'm wondering if we should have a full usage example based on pip (similar to what we have with [Go](https://github.com/containerbuildsystem/cachi2/blob/main/docs/usage.md)). Definitely not something to cover in this PR, though.

+1, already working on it :slightly_smiling_face:

**Review by ejegrova — COMMENTED** [review]
nitpicking :) Astonishing work, thank you.

File: docs/pip.md [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+├── ...
+├── pdm-pep517-1.0.6.tar.gz
+├── requests-2.28.2.tar.gz
+├── ...
+└── wheel-0.38.4.tar.gz
+```
+
+To make pip use the downloaded archives, use the [--find-links](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-f)
+and [--no-index](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-no-index) options. The --find-links option
+tells pip to looks for dependency archives in a directory, --no-index prevents pip from preferring PyPI over the local
```
[the remark this answers is not in the document]
Comment (chmeliik) [reply docs/pip.md]: Good catch, fixed :+1:

**Comment by brunoapimentel** [comment]
Nitpick: I really like the dropdowns you added in the text, but my first impression was that they were a little confusing. I tried to use some more sentences to make more clear what they are representing there: https://github.com/chmeliik/cachi2/compare/document-pip...brunoapimentel:cachi2:document-pip?diff=split

Feel free to change the suggestions here or just ignore them altogether.

File: docs/pip.md [hunk 2]
```diff
@@ -1,264 +1,571 @@
...
+
+```text
+cachi2-output/deps/pip
+├── ...
+├── external-dockerfile-parse
+│   └── dockerfile-parse-external-sha256-36e4469abb0d96b0e3cd656284d5016e8a674cd57b8ebe5af64786fe63b8184d.tar.gz
+└── ...
+```
+
+## Caveats, workarounds
```
Comment (brunoapimentel) [inline docs/pip.md:494]: I think this session ended up being just `workarounds`. :laughing: 

Maybe we can rename it to `Build troubleshooting` or something in this sense, and add a brief intro paragraph like: 
```
This section shows some common errors that can happen when attempting to hermetically build a container using Cachi2's pip prefetched content.
```

**Comment by chmeliik** [comment]
> Nitpick: I really like the dropdowns you added in the text, but my first impression was that they were a little confusing. I tried to use some more sentences to make more clear what they are representing there: https://github.com/chmeliik/cachi2/compare/document-pip...brunoapimentel:cachi2:document-pip?diff=split
> 
> Feel free to change the suggestions here or just ignore them altogether.

Applied some of the formatting changes from your diff. For making the dropdowns clearer, I'll play with adding introductory sentences vs. just extending the `<summary>`

**Comment by chmeliik** [comment]
> Applied some of the formatting changes from your diff. For making the dropdowns clearer, I'll play with adding introductory sentences vs. just extending the `<summary>`

I guess the problem is that users may not realize they can click to expand them, in which case a better summary won't help

**Comment by chmeliik** [comment]
I could also get rid of the dropdowns and show the example files right away (except for the big ones like requirements.txt)

File: docs/pip.md (hunk 2 above)
Comment (chmeliik) [reply docs/pip.md]: Renamed to Troubleshooting and added a short intro. It doesn't apply just to builds, for example https://github.com/chmeliik/cachi2/blob/document-pip/docs/pip.md#dependency-does-not-distribute-sources fails during prefetch already

**Comment by brunoapimentel** [comment]
> Applied some of the formatting changes from your diff. For making the dropdowns clearer, I'll play with adding introductory sentences vs. just extending the `<summary>`

So, you didn't like the results of adding introductory sentences? I'm fine with keeping the formatting as it is now.

**Review by brunoapimentel — APPROVED** [review]

**Comment by chmeliik** [comment]
> > Applied some of the formatting changes from your diff. For making the dropdowns clearer, I'll play with adding introductory sentences vs. just extending the `<summary>`
> 
> So, you didn't like the results of adding introductory sentences? I'm fine with keeping the formatting as it is now.

I didn't see much value in explaining that the examples are examples :smile: 

That said, I did restructure some things and add "Example: " to the summaries, let me know if that helps

**Review by lkolacek — APPROVED** [review]
Awesome work :+1:

**Review by ejegrova — APPROVED** [review]

**Pull request merged**